test/cmdreader.go

```diff
@@ -0,0 +1,79 @@
+package test
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os/exec"
+	"time"
+)
+
+const (
+	// DefaultTimeout for operations that are expected to succeed.
+	DefaultTimeout = 5 * time.Minute
+	// ShortTimeout for operations that are *expected* to time out.
+	ShortTimeout = 10 * time.Second
```

Comment:
I suggest to rename `DefaultTimeout` to `DefaultSuccessTimeout` and `ShortTimeout` to `DefaultFailureTimeout`

Reply:
I like it.

Reply:
I despise the very notion of a "DefaultFailureTimeout" but I haven't figured out how to eliminate it yet. It's "wishful thinking" testing - "well I didn't look very hard and I didn't find anything, so it must not be there."